docs: Add FAQ question about using the same machine as server and client (#1854)
juanfont/headscale: docs/faq.md

## Why is my reverse proxy not working with Headscale?

We don't know. We don't use reverse proxies with `headscale` ourselves, so we don't have any experience with them. We have [community documentation](https://headscale.net/reverse-proxy/) on how to configure various reverse proxies, and a dedicated [Discord channel](https://discord.com/channels/896711691637780480/1070619818346164324) where you can ask for help to the community.

> ## Can/Should I use the same machine as both server and client?

Reviewer: ```suggestion
## Can I use headscale and tailscale on the same machine?
```

> Running headscale on a machine that is also in the tailnet might cause problems with, at least, routing and MagicDNS. It should be extremely discouraged.

Reviewer 2: ```suggestion
Running headscale on a machine that is also in the tailnet can cause problems with subnet routers, traffic relay nodes, and MagicDNS. It might work, but it is not supported.
```
